dpcpp/preconditioner/batch_block_jacobi.hpp.inc

```diff
@@ -44,25 +44,83 @@ public:
                   const batch::matrix::ell::batch_item<const value_type,
                                                        const index_type>&,
                   value_type* const, sycl::nd_item<3> item_ct1)
-    {}
+    {
+        common_generate_for_all_system_matrix_types(batch_id);
+        item_ct1.barrier(sycl::access::fence_space::local_space);
```

Comment:
different sync place from cuda hip?

Reply:
I guess it is not really necessary, but I think it should not hurt.

Reply:
the barrier should take quite some overhead.
If it is unnecessary, maybe remove it.

Reply:
I would prefer to keep it for now. Will look into removing the unnecessary syncs at a later point.